pg-ragtime/src/pg/ragtime.clj

```diff
@@ -0,0 +1,29 @@
+(ns pg.ragtime
+  (:require
+   [ragtime.core :as ragtime]
+   [ragtime.strategy]
+   [ragtime.protocols]
+   [pg.core :as pg])
+  (:import org.pg.Connection))
+
+(defn ensure-migrations-table!
+  [conn]
+  (pg/query conn "
+create table if not exists ragtime_migrations(
+  id text primary key,
+  timestamp timestamp default current_timestamp
+)"))
+
+(defn dangerously-delete-migrations-table!!
+  [conn]
+  (pg/query conn "drop table ragtime_migrations"))
+
+(extend-type Connection
+  ragtime.protocols/DataStore
+  (add-migration-id [conn id]
+    (pg/execute conn "insert into ragtime_migrations(id) values ($1)" {:params [id]}))
+  (remove-migration-id [conn id]
+    (pg/execute conn "delete from ragtime_migrations where id = $1", {:params [id]}))
+  (applied-migration-ids [conn]
+    (->> (pg/query conn "select id from ragtime_migrations")
+         (map :id))))
```

Comment:
Could you also create 2-3 test migrations please and put them into the tests/migrations directory? Something like create a table `users` and then adding a column. Like this: https://github.com/weavejester/ragtime/tree/master/next-jdbc/test/migrations

Then, we can prepare tests that ensure migrations work.

Reply:
I can, definitely agree that this needs to be tested.

I haven't been able to run the code yet locally. I've never worked in a project like this where there are multiple `project.clj` files, dependency versions controlled _one place_, and Java source files. So I might have to spend some time understanding how to use Leiningen, a REPL and Java source files in this context.